test/UnitTest/Microsoft.AspNet.OData.Test/PublicApi/Microsoft.AspNet.OData.PublicApi.bsl

```diff
@@ -2296,9 +2321,24 @@ public class Microsoft.AspNet.OData.Query.DefaultQuerySettings {
 	bool EnableFilter  { public get; public set; }
 	bool EnableOrderBy  { public get; public set; }
 	bool EnableSelect  { public get; public set; }
+	bool EnableSkipToken  { public get; public set; }
 	System.Nullable`1[[System.Int32]] MaxTop  { public get; public set; }
 }
 
+public class Microsoft.AspNet.OData.Query.DefaultSkipTokenHandler : SkipTokenHandler {
+	public DefaultSkipTokenHandler ()
+	public DefaultSkipTokenHandler (char delimiter)
+
+	public virtual IQueryable`1 ApplyTo (IQueryable`1 query, SkipTokenQueryOption skipTokenQueryOption)
+	public virtual System.Linq.IQueryable ApplyTo (System.Linq.IQueryable query, SkipTokenQueryOption skipTokenQueryOption)
+	protected static System.Linq.IQueryable ApplyToCore (System.Linq.IQueryable query, ODataQuerySettings querySettings, System.Collections.Generic.IList`1[[Microsoft.AspNet.OData.Query.OrderByNode]] orderByNodes, ODataQueryContext context, string skipTokenRawValue)
+	public virtual System.Uri GenerateNextPageLink (System.Uri baseUri, int pageSize, object instance, ODataSerializerContext context)
+	protected virtual string GenerateSkipTokenValue (object lastMember, Microsoft.OData.Edm.IEdmModel model, System.Collections.Generic.IList`1[[Microsoft.AspNet.OData.Query.OrderByNode]] orderByNodes)
+	protected static System.Collections.Generic.IEnumerable`1[[Microsoft.OData.Edm.IEdmProperty]] GetPropertiesForSkipToken (object lastMember, Microsoft.OData.Edm.IEdmModel model, System.Collections.Generic.IList`1[[Microsoft.AspNet.OData.Query.OrderByNode]] orderByNodes)
+	protected static Microsoft.OData.Edm.IEdmType GetTypeFromObject (object value, Microsoft.OData.Edm.IEdmModel model)
+	protected static System.Collections.Generic.IDictionary`2[[System.String],[System.Object]] PopulatePropertyValuePairs (string value, ODataQueryContext context)
+}
```

Comment:
Do we need all of these protected, rather than private? Do intend for these to be overrideable in a derived class? Which can I reasonably replace, versus which are specific/dependent logic on this implementation?

Reply:
The initial thought was to reduce duplication of code if any of the methods can be reused by the subclass but it seems like only GenerateSkipTokenvValue should be the pluggable method.

---
In reply to: [287522787](https://github.com/OData/WebApi/pull/1724#discussion_r287522787) [](ancestors = 287522787)